src/SprykerEco/Zed/Heidelpay/Business/Order/Saver.php

```diff
@@ -122,10 +132,24 @@ protected function hydratePaymentMethodSpecificDataToPayment(
     /**
      * @param string $paymentMethodCode
      *
-     * @return boolean
+     * @return bool
      */
     protected function hasPaymentMethodSpecificDataToSave($paymentMethodCode)
     {
         return isset($this->paymentCollection[$paymentMethodCode]);
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Orm\Zed\Heidelpay\Persistence\SpyPaymentHeidelpay $paymentEntity
+     *
+     * @return void
+     */
+    protected function addBasketInformation(QuoteTransfer $quoteTransfer, SpyPaymentHeidelpay $paymentEntity): void
+    {
+        $heidelpayBasketResponseTransfer = $this->basketHandler
```

Comment:
Please remove this method from this method - it does 2 things - creates basket and stores ID into payment.